### ISSUE TITLE
Phyton

### ISSUE THREAD

**Issue by shaikzakeer902-sys** [shaikzakeer902-sys/Samg-ai-agent#2]
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler, OneHotEncoder
from sklearn.compose import ColumnTransformer
from sklearn.pipeline import Pipeline

class DataAgent:
    def __init__(self, target_column, test_size=0.2, random_state=42):
        self.target = target_column
        self.test_size = test_size
        self.random_state = random_state
        self.pipeline = None

    def load_csv(self, path):
        df = pd.read_csv(path)
        return df

    def basic_clean(self, df):
        # Simple cleaning: drop exact duplicate rows
        df = df.drop_duplicates().reset_index(drop=True)
        return df

    def build_pipeline(self, X):
        # infer numerical and categorical
        numeric_cols = X.select_dtypes(include=['int64', 'float64']).columns.tolist()
        cat_cols = X.select_dtypes(include=['object', 'category', 'bool']).columns.tolist()

        numeric_transformer = Pipeline(steps=[('scaler', StandardScaler())])
        categorical_transformer = Pipeline(steps=[('ohe', OneHotEncoder(handle_unknown='ignore'))])

        preprocessor = ColumnTransformer(transformers=[
            ('num', numeric_transformer, numeric_cols),
            ('cat', categorical_transformer, cat_cols)
        ], remainder='drop')

        self.pipeline = preprocessor
        return preprocessor

    def preprocess(self, df):
        df = self.basic_clean(df)
        X = df.drop(self.target, axis=1)
        y = df[self.target]

        preprocessor = self.build_pipeline(X)
        X_trans = preprocessor.fit_transform(X)

        return train_test_split(X_trans, y, test_size=self.test_size, random_state=self.random_state)
